### DISCUSSION TITLE
无状态服务器安装指导

### DISCUSSION THREAD

**Discussion by huoxingdeguoguo** [chanify/chanify#76] [Q&A]
使用docker 运行时命令如下docker run -it wizjin/chanify:latest serve --secret=<secret key> --name=<node name> --endpoint=http://<address>:<port>，有大神知道endpoin怎么填写吗？是服务器地址绑定的域名吗？端口号只要服务器开放的无冲突的就可以吗？谢谢了

**Comment by wizjin** [comment]
endpoint 是手机 app 链接服务器用的，可以是域名，也可以是 ip 地址。如果没有前置的 web 反向代理，端口号只要服务器开放的无冲突就可以了，否则要看前置 web 反向代理的配置。

**Reply by huoxingdeguoguo** [reply]
首先谢谢这么快解答，我还有一点不明白，我服务器运行chanify的无状态服务是不是这个服务器就不能作为发送端了，只能再用另外一台服务器安装chanify作为发送端？

**Reply by wizjin** [reply]
发送是用的http接口，你可以使用诸如 curl 等工具发送，也可以在自己编写脚本或者代码。或者支持 webhook 触发的也可以用作发送。这个和是不是无状态服务没关系。无状态服务只是不保存用户的 device token，有状态的可以保存 device token 和部分数据。所以如果要发送图片或者文件的话只能用有状态的。

**Reply by huoxingdeguoguo** [reply]
懂了，感谢解答
